### PR TITLE
Add spelling exclusions file

### PR DESCRIPTION
**Pull request by MiYanni**
## Summary
In my investigation of VS squiggles for the repo, there were a number of them for spelling issues. I've added a `.dic` file that includes a list of words that can be treated as spelling exclusions. Details on the exclusions file can be [found here](https://learn.microsoft.com/en-us/visualstudio/ide/text-spell-checker?view=vs-2022#how-to-customize-the-spell-checker).

I couldn't figure out how to investigate **all** spelling issues for the SDK. Instead, I just flipped through a bunch of files and found random spelling mistakes that I could either fix or add to the exclusions list.

I encourage others to add more words to the `exclusion.dic` as they run into them.

### CODE REVIEW

**Comment by danmoseley** [comment]
@MiYanni  how is the spell checker hooked up - is it the default thing I get in VS?

**Comment by danmoseley** [comment]
I wonder whether there's a way we can share a suppression dictionary across our repos through Arcade?

Is there any way to run it on the command line?

**Comment by Forgind** [comment]
> I wonder whether there's a way we can share a suppression dictionary across our repos through Arcade?

It looks like this was added through the editorconfig, so each repo would have to add it to their own editorconfig, but then Arcade could provide a exclusion dictionary for all repos.

> Is there any way to run it on the command line?

Is there a way to make the default spell checker run on the command line? I thought it wouldn't catch any spelling mistakes outside VS, and we'd need that before we can worry about an exclusion list.

**Review by Forgind — APPROVED** [review]
Love it!

**Comment by MiYanni** [comment]
@danmoseley 
> how is the spell checker hooked up - is it the default thing I get in VS?

[Here](https://learn.microsoft.com/en-us/visualstudio/ide/text-spell-checker?view=vs-2022) is the article on the spell checker. It was added in VS in 17.5. AFAIK, it is on by default. [Here](https://learn.microsoft.com/en-us/visualstudio/ide/text-spell-checker?view=vs-2022#how-to-customize-the-spell-checker) is the section of the article on the spell checker for this exclusion setting.

> I wonder whether there's a way we can share a suppression dictionary across our repos through Arcade?

Since the exclusion setting is just a path, you could set it to `.\eng\common\exclusions.dic` for example. However, the setting it put in the `.editorconfig` which is normally at the root of the repo. So yeah, just mirroring what Nathan said about it.

> Is there any way to run it on the command line?

Based on [this](https://learn.microsoft.com/en-us/visualstudio/ide/text-spell-checker?view=vs-2022#how-does-it-work), I don't think so:
> [!Note]
> Feedback from early users of this feature informed us that developers only want to know about errors in the document they are currently working with. In response to that feedback, the spell checker only scans the document(s) that are active in the editor.

My intuition would lead me to think this would be a feature that would need to be added to `dotnet format` if we wanted a commandline version.

### Toward your Arcade proposal
If there was a way to run it for an entire solution, compiling a list of common exclusions would be easily possible. The best we could do right now is to do a GitHub search for `spelling_exclusion_path` (which is quite unique) and then aggregate many of the `.dic` files and compile a list manually. That's what I'd do if I was making a shared, multi-repo dictionary file (like one Arcade would distribute).

File: exclusion.dic [hunk 1]
```diff
@@ -0,0 +1,14 @@
+awaiter
+crossgen
+cshtml
+deserializer
+nuget
+nupkg
+nuspec
+sdk
+sdks
```
Comment (nagilson) [inline exclusion.dic:9]:
```suggestion
sdks
msbuild
```

**Review by nagilson — APPROVED** [review]
This looks awesome, thanks @MiYanni  :)

File: exclusion.dic (hunk 1 above)
Comment (MiYanni) [reply exclusion.dic:9]: I'll add it in alphabetical order. One sec.

File: src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/SuppressibleMSBuildLog.cs [hunk 2]
```diff
@@ -7,16 +7,16 @@
 namespace Microsoft.DotNet.ApiCompat.Task
 {
     /// <summary>
-    /// Class that can log Suppressions in an MSBuild task, by implementing MSBuildLog and ISuppressableLog.
+    /// Class that can log Suppressions in an MSBuild task, by implementing MSBuildLog and ISuppressibleLog.
     /// </summary>
-    internal sealed class SuppressableMSBuildLog : MSBuildLog, ISuppressableLog
+    internal sealed class SuppressibleMSBuildLog : MSBuildLog, ISuppressableLog
```
Comment (danmoseley) [inline src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/SuppressibleMSBuildLog.cs:12]:
```suggestion
    internal sealed class SuppressibleMSBuildLog : MSBuildLog, ISuppressibleLog
```
?

Comment (MiYanni) [reply src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/SuppressibleMSBuildLog.cs:12]: Oh golly. Okay. Let me get all those too.

Comment (MiYanni) [reply src/ApiCompat/Microsoft.DotNet.ApiCompat.Task/SuppressibleMSBuildLog.cs]: Fixed all the uses of 'suppressable' in the repo.

**Review by ViktorHofer — APPROVED** [review]
Approving the APICompat changes.

**Comment by MiYanni** [comment]
Note: This is currently bugged in some recent versions of VS. It isn't working for me currently on my desktop machine, and I wanted to know why. So, if you experience the same thing, we'll just have to wait for whatever VS version has this fix: https://developercommunity.visualstudio.com/t/VS2022-1752-Spell-Checker-Exclusion-Di/10328416

**Pull request merged**